pkg/kubectl/cmd/create_role.go

```diff
@@ -147,10 +166,17 @@ func (c *CreateRoleOptions) Validate(f cmdutil.Factory) error {
 	}
 
 	for _, r := range c.Resources {
-		_, err := mapper.ResourceFor(r)
-		if err != nil {
+		if len(r.Resource) == 0 {
+			return fmt.Errorf("resource must be specified if apiGroup/subresource specified")
+		}
+		if _, err := mapper.ResourceFor(schema.GroupVersionResource{Resource: r.Resource, Group: r.Group}); err != nil {
 			return err
 		}
+		if len(r.SubResource) > 0 {
+			if _, err := mapper.ResourceFor(schema.GroupVersionResource{Resource: r.SubResource}); err != nil {
```

Comment:
I think you have to take the resolved resource, then append the subresource you want:

```
resource, err := mapper.ResourceFor(schema.GroupVersionResource{Resource: r.Resource, Group: r.Group})
...
if len(r.SubResource) > 0 {
  subresource, err := mapper.ResourceFor(schema.GroupVersionResource{
    Group: r.Group
    Resource: resource.Resource + "/" + r.SubResource,
  })
  ...
}
```